src/main/java/Parser.java

```diff
@@ -0,0 +1,109 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+/**
+ * Represents the user input Parser handler.
+ */
+public class Parser {
+    private static final Ui ui = new Ui();
+
+    /**
+     * Processes the user input date and returns it in the correct format.
+     * @param s The user input date
+     * @return The formatted date for use.
+     */
+    public static String parseDateTime(String s) {
+        SimpleDateFormat formatter = new SimpleDateFormat("dd MMM yyyy h:mma");
+        SimpleDateFormat parser = new SimpleDateFormat("yyyy-MM-dd HHmm");
+        SimpleDateFormat dateFormatter = new SimpleDateFormat("dd MMM yyyy");
+        SimpleDateFormat dateParser = new SimpleDateFormat("yyyy-MM-dd");
+
+        try {
+            if (s.contains(" ")) {
+                return formatter.format((parser.parse(s)));
+            } else {
+                return dateFormatter.format(dateParser.parse(s));
+            }
+        } catch (ParseException e) {
+            return "Please input the time and date in\n"
+                + dateParser.toPattern() + " or " + parser.toPattern() + "\n";
+        }
+    }
+
+    public static boolean checkBye(String s) {
```

Comment:
Since there are so many similar methods, do consider extracting them into a separate `CommandChecker` class. I think that would help organize the commands. The current setup might not scale as well if there are more valid inputs that you need to check against.

An alternative might be to use a single `checkCommand(String s)` method that returns an enum. This solution would allow you to switch on an enum instead of a boolean, which might enhance the readability of the `parse(TaskList taskList, Storage storage, String text)` method.